src/server/handlers.rs

```diff
@@ -1198,4 +1256,28 @@ mod test {
         assert!(!regex.is_match(word_with_period));
         assert!(!regex.is_match(word_with_space));
     }
+
+    #[tokio::test]
+    async fn validate_oam_label_rules() {
+        // Valid labels
+        assert!(valid_oam_label((&"foo".to_string(), &"bar".to_string())));
+        assert!(valid_oam_label((
+            &"app.oam.io/name".to_string(),
+            &"wasmcloud".to_string()
+        )));
+        assert!(valid_oam_label((
+            &"justaregularstring".to_string(),
+            &"wasmcloud".to_string()
+        )));
+
+        // Invalid labels
```

Comment:
Do we need some checking for whitespace?

Reply:
Yeah will be good to add a test for this